SourceCode/CMakeLists.txt

```diff
@@ -2,8 +2,9 @@ cmake_minimum_required(VERSION 3.19)
 project(Chaos)
 set(CMAKE_CXX_STANDARD 20)
 
-set(SOURCE_FILES src/color.cpp src/camera.cpp src/ray.cpp src/triangle.cpp src/scene.cpp src/renderer.cpp
-    src/random_generator.cpp src/image_factory.cpp src/scene_factory.cpp src/stb_image_write.cpp)
+set(SOURCE_FILES src/camera.cpp src/ray.cpp src/triangle.cpp src/geometry.cpp src/scene.cpp src/renderer.cpp src/color.cpp
+    src/random_generator.cpp src/image_factory.cpp src/scene_factory.cpp src/scene_builder.cpp
+    src/file_handler.cpp src/stb_image_write.cpp)
```

Comment:
A very minor thing, but you will have cleaner git history, as well as lower chance of merge conflicts if you put each source on a separate line.